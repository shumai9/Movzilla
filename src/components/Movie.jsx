import React, { Component } from 'react';
import Screen from './Screen';
import '../style/Movie.css';

const Api_key = `${process.env.REACT_APP_OMDb_API_KEY}`;
const API_URL = 'http://www.omdbapi.com/?'

class Movie extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loaded: false,
      data: null,
      count: 0,
      next: null,
      featured: this.props.featured
    }
  }

  clearSearch =(e)=>{
    e.target.value = ''
    return <p>"please write title to search"</p>
  }
  
  handleSearch=(e)=>{
    e.preventDefault();    
    let req = document.querySelector('.search').value;

    if(req) {     
      fetch(`${API_URL}s=${req}&type=movie&page=1&apikey=${Api_key}`)
      .then(res => res.json())
      .then(result => {
        if(result.Response){
          this.setState({
            data: result.Search,
            count: result.totalResults,
            loaded: true,
            next:''
          })
          console.log("From fetch", req)
        }
      })
      .catch(
        e => {console.log("From movie", e);}
      );
    }else{
      return
    }
    console.log('from handler', req);
  }

  render(){
    const clearSelected = this.props.clearSelected;
    const movie= this.props.movie;
    const featured= this.props.featured;
  
    return(
      <div className="movies">
        <h1>Movies</h1>
        <div>
        <form onSubmit={this.handleSearch}>
          <input 
            className="search title"
            value={this.state.search_bar}
            onChange={this.handleSearch}
            onFocus={this.clearSearch}
            type="text"  
            name="search_bar"
            placeholder="type movie title here"
          />
          <button type="submit"> Search </button>
          </form>
          <p>Total search results: {this.state.count}</p>
        </div>
        <div className="wrapper" >
          { this.state.data ? 
            (Object.values(this.state.data).map((line) =>{
              return(
                <div className="film" key={line.imdbID}> 
                  <h2 >{line.Title}</h2>
                  <img 
                    src={line.Poster}
                    alt={`${line.Title}`}
                  />          
                  <p>{line.Year}</p> 
                  <p>{line.Type}</p>
                </div>
              ) 
            })
            ) : (
              <Screen
                movie={movie}
                featured={featured} 
                clearSelected={clearSelected}
              />
            )
          }
        </div>      
      </div>
    );
  }
}

export default Movie;

