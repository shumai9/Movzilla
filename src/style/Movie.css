.movies{
  width: auto;
}
.movies form, .newly p{
  z-index: 100;
  position: absolute;
  right: 1rem;
  top: 5rem;
  display: flex;
}
.main, .movie-list {
  flex: 1 2 auto;
  background: rgba(156, 146, 146, 0.28);
  border: 1px solid #000;
  padding: 10px;
  margin: 10px;
}

.wrapper{
  align-items: center;
  display: grid;
  max-width: 100%;
  grid-template-columns: repeat(auto-fit, minmax(16rem, 1fr));
  grid-template-rows: repeat(3, auto);
  grid-gap: 15px 15px;
  padding: 15px;
}
div.wrapper img {
  border-radius: 1rem;
}
.screen {
  width: auto;
  background: #000;
}