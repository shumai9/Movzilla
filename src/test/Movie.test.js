import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Movie from '../components/Movie';

it('renders with out  failing', () =>{
  const div = document.createElement('div');
  ReactDOM.render(<Movie />, div);
  ReactDOM.unmountComponentAtNode(div);
});
it('renders correctly', () =>{
  const tree = renderer
    .create(<Movie />)
    .toJSON();
  expect(tree).toMatchSnapshot();
}); 

