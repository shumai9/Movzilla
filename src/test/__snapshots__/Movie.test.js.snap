// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="movies"
>
  <h1>
    Movies
  </h1>
  <div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="search title"
        name="search_bar"
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="type movie title here"
        type="text"
      />
      <button
        type="submit"
      >
         Search 
      </button>
    </form>
    <p>
      Total search results: 
      0
    </p>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="screen"
      onClick={[Function]}
    />
  </div>
</div>
`;
